Test suite is unstable, sometimes RrdpServiceImplTest fails on master

Tests in `RrdpServiceImplTest`  fail often (6 out of last 7 runs) fails on my machine on master.

```
❯ java -version
openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment (build 11.0.7+10)
OpenJDK 64-Bit Server VM (build 11.0.7+10, mixed mode)
```

```
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR]   RrdpServiceImplTest.should_parse_notification_use_delta_mismatching_delta_hash_fallback_to_snapshot:540 expected:<1> but was:<18862>
[INFO]
[ERROR] Tests run: 122, Failures: 1, Errors: 0, Skipped: 6
...
[ERROR] Failures:
[ERROR]   RrdpServiceImplTest.should_parse_notification_and_snapshot:147 expected:<2> but was:<1567>
[INFO]
[ERROR] Tests run: 122, Failures: 1, Errors: 0, Skipped: 6
[INFO]
...
[ERROR] Failures:
[ERROR]   RrdpServiceImplTest.should_parse_and_save_snapshot:97 expected:<3> but was:<18864>
[INFO]
[ERROR] Tests run: 122, Failures: 1, Errors: 0, Skipped: 6
...
[ERROR] Failures:
[ERROR]   RrdpServiceImplTest.should_parse_notification_use_delta_and_republish_object:380 expected:<1> but was:<1566>
[INFO]
[ERROR] Tests run: 122, Failures: 1, Errors: 0, Skipped: 6
```

--- a/rpki-validator/src/main/java/net/ripe/rpki/validator3/api/health/HealthController.java
+++ b/rpki-validator/src/main/java/net/ripe/rpki/validator3/api/health/HealthController.java
@@ -45,6 +45,7 @@ import net.ripe.rpki.validator3.background.BackgroundJobs;
 import net.ripe.rpki.validator3.storage.Storage;
 import net.ripe.rpki.validator3.storage.stores.TrustAnchors;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
@@ -59,6 +60,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
+@Profile("!test")
 @InternalApiCall
 @RestController
 @RequestMapping(path = "/api/healthcheck", produces = {ValidatorApi.API_MIME_TYPE, "application/json"})

--- a/rpki-validator/src/main/java/net/ripe/rpki/validator3/background/BackgroundJobs.java
+++ b/rpki-validator/src/main/java/net/ripe/rpki/validator3/background/BackgroundJobs.java
@@ -44,6 +44,7 @@ import org.quartz.listeners.JobListenerSupport;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.annotation.Profile;
 import org.springframework.context.event.EventListener;
 import org.springframework.stereotype.Component;
 
@@ -62,6 +63,7 @@ import static org.quartz.JobBuilder.newJob;
 import static org.quartz.SimpleScheduleBuilder.simpleSchedule;
 import static org.quartz.TriggerBuilder.newTrigger;
 
+@Profile("!test")
 @Component
 @Slf4j
 public class BackgroundJobs extends JobListenerSupport {

--- a/rpki-validator/src/main/java/net/ripe/rpki/validator3/background/ValidationScheduler.java
+++ b/rpki-validator/src/main/java/net/ripe/rpki/validator3/background/ValidationScheduler.java
@@ -30,15 +30,12 @@
 package net.ripe.rpki.validator3.background;
 
 import com.google.common.base.Preconditions;
-import lombok.AllArgsConstructor;
-import lombok.Data;
 import lombok.Getter;
 import lombok.extern.slf4j.Slf4j;
 import net.ripe.rpki.validator3.api.ValidatorApi;
 import net.ripe.rpki.validator3.domain.validation.CertificateTreeValidationService;
 import net.ripe.rpki.validator3.storage.data.RpkiRepository;
 import net.ripe.rpki.validator3.storage.data.TrustAnchor;
-import org.quartz.JobKey;
 import org.quartz.Scheduler;
 import org.quartz.SchedulerException;
 import org.quartz.SimpleScheduleBuilder;
@@ -46,17 +43,12 @@ import org.quartz.TriggerBuilder;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Lazy;
+import org.springframework.core.env.Environment;
+import org.springframework.core.env.Profiles;
 import org.springframework.dao.EmptyResultDataAccessException;
 import org.springframework.stereotype.Component;
 
 import java.time.Duration;
-import java.time.Instant;
-import java.time.temporal.TemporalField;
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.TimeUnit;
 
 @Component
 @Slf4j
@@ -79,7 +71,8 @@ public class ValidationScheduler {
     public ValidationScheduler(Scheduler scheduler,
                                @Value("${rpki.validator.rsync.repository.download.interval:PT10M}") String rsyncRepositoryDownloadInterval,
                                @Value("${rpki.validator.rrdp.repository.download.interval:PT2M}") String rrpdRepositoryDownloadInterval,
-                               @Lazy CertificateTreeValidationService validationService) {
+                               @Lazy CertificateTreeValidationService validationService,
+                               Environment environment) {
         this.scheduler = scheduler;
         this.rsyncRepositoryDownloadInterval = Duration.parse(rsyncRepositoryDownloadInterval);
         this.rrpdRepositoryDownloadInterval = Duration.parse(rrpdRepositoryDownloadInterval);
@@ -93,6 +86,11 @@ public class ValidationScheduler {
                 this.rrpdRepositoryDownloadInterval.getSeconds()));
 
         throttledTreeValidation = new Throttled<>(minIntervalBetweenTreeValidationsInSeconds);
+
+        // Disable scheduling during tests
+        if (environment.acceptsProfiles(Profiles.of("test"))) {
+            this.disable();
+        }
     }
 
     public void addTrustAnchor(TrustAnchor trustAnchor) {

